src/D2L.Security.OAuth2/Keys/KeyManagementService.cs

```diff
@@ -57,14 +70,11 @@ OAuth2Configuration config
 				await RefreshKeyAsync( now )
 					.ConfigureAwait( false );
 
-				current = Volatile.Read( ref m_current );
+				using var refreshed = GetCurrentToken();
+				return refreshed?.Ref();
 			}
 
-			if( current == null ) {
-				return null;
-			}
-
-			return current.Ref();

```

Comment:
this is the main race condition - if the key is replaced between `Volatile.Read` and this call to `.Ref()` is will be disposed.